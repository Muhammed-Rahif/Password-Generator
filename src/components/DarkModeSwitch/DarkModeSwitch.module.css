.darkModeBtnWrapper {
  position: absolute;
  top: 0.75rem;
  right: 0rem;
  display: flex;
  align-items: center;
}

.darkModeBtn {
}
