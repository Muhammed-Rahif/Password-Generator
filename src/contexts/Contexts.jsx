import React, { createContext, useState } from "react";

const isDeviceLightTheme =
  window.matchMedia && window.matchMedia("(prefers-color-scheme: light)").matches;

export const LightThemeContext = createContext(isDeviceLightTheme);

function Contexts({ children }) {
  const [lightTheme, setLightTheme] = useState(isDeviceLightTheme);

  return (
    <LightThemeContext.Provider value={{ lightTheme, setLightTheme }}>
      {children}
    </LightThemeContext.Provider>
  );
}

export default Contexts;
