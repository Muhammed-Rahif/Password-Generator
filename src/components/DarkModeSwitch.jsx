import React, { useContext } from "react";
import { Subtitle2, Switch } from "ui-neumorphism";
import { LightThemeContext } from "../contexts/Contexts";
import styles from "./DarkModeSwitch.module.css";

function DarkModeSwitch(props) {
  const { lightTheme, setLightTheme } = useContext(LightThemeContext);

  const toggleTheme = e => {
    setLightTheme(!e.target.checked);
  };

  return (
    <div className={styles.darkModeBtnWrapper}>
      <Subtitle2 dark={!lightTheme}>Dark Mode</Subtitle2>
      <Switch dark={!lightTheme} checked={!lightTheme} onClick={toggleTheme} />
    </div>
  );
}

export default DarkModeSwitch;
