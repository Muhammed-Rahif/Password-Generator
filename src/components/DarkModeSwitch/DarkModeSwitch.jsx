import React, { useContext } from "react";
import { Subtitle2, Switch } from "ui-neumorphism";
import { LightThemeContext } from "../../contexts/Contexts";
import styles from "./DarkModeSwitch.module.css";

function DarkModeSwitch(props) {
  const { lightTheme, setLightTheme } = useContext(LightThemeContext);

  const toggleTheme = darkMode => {
    setLightTheme(!darkMode);
  };

  return (
    <div className={styles.darkModeBtnWrapper}>
      <Subtitle2 dark={!lightTheme}>Dark Mode</Subtitle2>
      <Switch
        dark={!lightTheme}
        checked={!lightTheme}
        onClick={e => toggleTheme(e.target.checked)}
      />
    </div>
  );
}

export default DarkModeSwitch;
