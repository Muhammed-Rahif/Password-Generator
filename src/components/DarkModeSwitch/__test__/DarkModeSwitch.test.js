import React from "react";
import renderer from "react-test-renderer";
import Contexts from "../../../contexts/Contexts";
import DarkModeSwitch from "../DarkModeSwitch";

describe("dark mode switch", () => {
  const darkModeSwitchComponent = renderer.create(
    <Contexts>
      <DarkModeSwitch />
    </Contexts>
  );
  let componentTree = darkModeSwitchComponent.toJSON();

  it("should dark mode switch render without crashing", () => {
    expect(componentTree).toMatchSnapshot();
  });

  test("should change light theme context to true when dark mode switch is disabled", () => {
    darkModeSwitchComponent.root
      .findByType("input")
      .props.onClick({ target: { checked: false } });
    let componentTreeWhenDarkMode = darkModeSwitchComponent.toJSON();
    expect(componentTreeWhenDarkMode).toMatchSnapshot();
  });
});
