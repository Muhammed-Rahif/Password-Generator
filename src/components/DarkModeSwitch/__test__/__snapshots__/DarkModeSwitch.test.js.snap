// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dark mode switch should change light theme context to true when dark mode switch is disabled 1`] = `
<div
  className="darkModeBtnWrapper"
>
  <p
    className="_3yPwv _2vNlX _11JjB _1uvOH "
    style={Object {}}
  >
    Dark Mode
  </p>
  <div
    className="_jzWXa "
    style={Object {}}
  >
    <input
      checked={true}
      className="_AQLzt _1zQ0s"
      id="_01177_"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <label
      className="_2hoJO _17_OS"
      htmlFor="_01177_"
    >
      <p
        className="_3yPwv _NwSHs _11JjB _1uvOH "
        style={Object {}}
      />
    </label>
  </div>
</div>
`;

exports[`dark mode switch should dark mode switch render without crashing 1`] = `
<div
  className="darkModeBtnWrapper"
>
  <p
    className="_3yPwv _2vNlX _11JjB _3PUqy "
    style={Object {}}
  >
    Dark Mode
  </p>
  <div
    className="_jzWXa "
    style={Object {}}
  >
    <input
      checked={true}
      className="_AQLzt _BQ4h8"
      id="_01177_"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <label
      className="_2hoJO _17_OS"
      htmlFor="_01177_"
    >
      <p
        className="_3yPwv _NwSHs _11JjB _3PUqy "
        style={Object {}}
      />
    </label>
  </div>
</div>
`;
